### PR TITLE
[SYCLomatic] Support migration of 5 new and 5 overload CUB API

### PR DESCRIPTION
**Pull request by yihwang-nv**
- cub::IADD3
- cub::WarpReduce::Reduce
- cub::DeviceReduce::ArgMin
- cub::DeviceReduce::ArgMax
- cub::DeviceSegmentedSort::SortPairs
- cub::DeviceSegmentedSort::SortPairsDescending
- cub::DeviceSegmentedSort::SortPairs (cub::DoubleBuffer overload version)
- cub::DeviceSegmentedSort::SortPairsDescending  (cub::DoubleBuffer overload version)
- cub::DeviceSegmentedRadixSort::SortPairs  (cub::DoubleBuffer overload version)
- cub::DeviceSegmentedRadixSort::SortPairsDescending  (cub::DoubleBuffer overload version)

Signed-off-by: Wang, Yihan [yihan.wang@intel.com](mailto:yihan.wang@intel.com)

### CODE REVIEW

**Review by AndyCHHuang — APPROVED** [review]
LGTM

File: clang/test/dpct/cub/intrinsic/shl_add.cu [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+// CHECK-NEXT:  *res = (((unsigned int)1 << (unsigned int)2) + (unsigned int)3);
+// CHECK-NEXT:}
+__global__ void kernel1(int *res) {
+  *res = cub::SHL_ADD(1, 2, 3);
+}
+
+// CHECK:void kernel2(int *res, unsigned a, int b, unsigned c) {
+// CHECK-NEXT:  *res = ((a << (unsigned int)b) + c);
+// CHECK-NEXT:}
+__global__ void kernel2(int *res, unsigned a, int b, unsigned c) {
```
Comment (zhimingwang36) [inline clang/test/dpct/cub/intrinsic/shl_add.cu]: there is clamp logic there of SHL_ADD,   need extra option on parameter B. 
asm ("vshl.u32.u32.u32.clamp.add %0, %1, %2, %3;" :
        "=r"(ret) : "r"(x), "r"(shift), "r"(addend));

For test purpose, suggest to add help function for SHL_ADD.

Comment (zhimingwang36) [reply clang/test/dpct/cub/intrinsic/shl_add.cu]: SHL/SHR have moved to a seperate PR.

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**